From 8da5b2ebf893af5710024e7111143f2521707647 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 2 May 2012 09:21:05 +0000
Subject: [PATCH] init: fix inittab for users who upgrades from pre openrc

The /etc/init.d/rc[SLK] have been replaced with openrc but users
might have references to the old scripts in their /etc/inittab.

We fix this from initramfs init.
---
 initramfs-init.in |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/initramfs-init.in b/initramfs-init.in
index bec3b52..4425990 100755
--- a/initramfs-init.in
+++ b/initramfs-init.in
@@ -490,6 +490,14 @@ if [ -f "$ovl" ]; then
 			break;
 		fi
 	done
+
+	# fix up inittab from pre openrc times (alpine v1.8)
+	if [ -f "$sysroot"/etc/inittab ]; then
+		sed -i -e 's|:/etc/init.d/rcS|:/sbin/rc sysinit|' \
+			-e 's|:/etc/init.d/rcL|:/sbin/rc default|' \
+			-e 's|:/etc/init.d/rcK|:/sbin/rc shutdown|' \
+			"$sysroot"/etc/inittab
+	fi
 else
 	# add some boot services by default
 	rc_add devfs sysinit
-- 
1.7.10

